src/editor/ui_menus/main_menu_map_options.cc

```diff
@@ -36,6 +36,146 @@ inline EditorInteractive& MainMenuMapOptions::eia() {
+	}
+	dd->set_tooltip(_("Add a player to this team"));
+	dd->selected.connect([this, dd]() {
+		const Widelands::PlayerNumber player = dd->get_selected();
+		// add this player to this team and remove him from all other teams in this lineup
+		for (Widelands::SuggestedTeam& t : team_) {
+			for (auto it = t.begin(); it != t.end(); ++it) {
+				if (*it == player) {
+					t.erase(it);
+					break;
+				}
+			}
+		}
+		// determine our index (it may have changed since the creation!)
+		int dd_index = -1;
+		for (size_t i = 0; i < dropdowns_.size(); ++i) {
+			if (dropdowns_[i] == dd) {
+				dd_index = i;
+				break;
+			}
+		}
+		assert(dd_index >= 0);
+		if (static_cast<unsigned>(dd_index) >= team_.size()) {
+			assert(static_cast<unsigned>(dd_index) == team_.size());
+			team_.push_back(Widelands::SuggestedTeam());
+		}
+		team_[dd_index].push_back(player);
+		update();
+	});
+	return dd;
+}
+
+void SuggestedTeamsEntry::layout() {
+	const uint16_t h = kSuggestedTeamsUnitSize * (team_.size() + 1);
+	set_desired_size(get_w(), h);
+	delete_.set_pos(Vector2i(get_w() / 2, h - kSuggestedTeamsUnitSize));
+	delete_.set_size(get_w() / 2, kSuggestedTeamsUnitSize);
+
+	size_t index = 0;
+	for (auto& dd : dropdowns_) {
+		dd->set_size(kSuggestedTeamsUnitSize, kSuggestedTeamsUnitSize);
+		dd->set_pos(Vector2i(0, kSuggestedTeamsUnitSize * (index++)));
+	}
+}
+
+// Delete empty teams, and append an empty team to the end if not present
+void SuggestedTeamsEntry::update() {
+	int nr_teams = team_.size();
+	int nr_dd = dropdowns_.size();
+	for (int i = 0; i < nr_teams;) {
+		if (team_[i].empty()) {
+			dropdowns_[i]->die();
+
+			for (int j = i + 1; j < nr_dd; ++j) {
+				dropdowns_[j - 1] = dropdowns_[j];
+			}
+			dropdowns_.resize(nr_dd - 1);
+			for (int j = i + 1; j < nr_teams; ++j) {
+				team_[j - 1] = team_[j];
+			}
+			team_.resize(nr_teams - 1);
+
+			--nr_teams;
+			--nr_dd;
+		} else {
+			++i;
+		}
+	}
+	assert(nr_teams >= 0);
+	assert(nr_dd >= 0);
+
+	assert((nr_dd == nr_teams) ^ (nr_dd == nr_teams + 1));
+	if (nr_dd <= nr_teams) {
+		dropdowns_.push_back(create_dropdown(nr_teams));
+	}
+
+	layout();
+}
+
+bool SuggestedTeamsEntry::handle_mousepress(uint8_t btn, int32_t x, int32_t y) {
+	const int row = y / kSuggestedTeamsUnitSize;
+	const int col = x / kSuggestedTeamsUnitSize;
+	if (row >= 0 && static_cast<unsigned>(row) < team_.size()) {
+		if (col > 0 && static_cast<unsigned>(col) <= team_[row].size()) {
+			team_[row].erase(team_[row].begin() + (col - 1));
+			update();
+			return true;
+		}
+	}
+
+	return UI::Panel::handle_mousepress(btn, x, y);
+}
+
+void SuggestedTeamsEntry::draw(RenderTarget& r) {
+	UI::Panel::draw(r);
+
+	for (size_t row = 0; row < team_.size(); ++row) {
+		for (size_t col = 0; col < team_[row].size(); ++col) {
+			r.blit(Vector2i((col + 1) * kSuggestedTeamsUnitSize, row * kSuggestedTeamsUnitSize),
+			       playercolor_image(team_[row][col], "images/players/player_position_menu.png"));
```

Comment:
You could not them flat UI:Buttons. They can then handle their own clicks.

Reply:
done